src/model/evaluate.py

```diff
@@ -21,57 +21,37 @@ def eval_model(model, loader: torch.utils.data.DataLoader, state: str, log: logg
     """
     model.eval()
     with torch.no_grad():
-        num_piece_classes = len(REVERSED_PIECE_TYPE)
-        num_piece_colors = len(REVERSED_PIECE_COLOR)
+        num_classes = len(LABELS)
 
-        type_correct_hits = torch.zeros(num_piece_classes)
-        type_counts = torch.zeros(num_piece_classes)
-        color_correct_hits = torch.zeros(num_piece_colors)
-        color_counts = torch.zeros(num_piece_colors)
+        class_correct_hits = torch.zeros(num_classes)
+        class_counts = torch.zeros(num_classes)
 
-        for i, (image, type_label, color_label, is_piece) in enumerate(loader):
-            type_pred_probs, color_pred_probs = model(image)
-            type_pred = torch.argmax(type_pred_probs, axis=1)
-            color_pred = torch.argmax(color_pred_probs, axis=1)
+        for i, (image, label) in enumerate(loader):
+            class_probs = model(image.to(device))
+            class_pred = torch.argmax(torch.exp(class_probs), axis=1).cpu()
+            image.cpu()
 
-            type_label = torch.argmax(type_label, axis=1)
-            color_label = torch.argmax(color_label, axis=1)
+            class_label = torch.argmax(label, axis=1)
 
-            type_correct_hits[type_label] += (type_pred == type_label).to(torch.int64)
-            labels_count_per_class = torch.bincount(type_label)
-            type_counts[labels_count_per_class.nonzero()] += labels_count_per_class[labels_count_per_class.nonzero()]
+            class_correct_hits[class_label] += (class_pred == class_label).to(torch.int64)
```

Comment:
This only makes sense in the `batch_size=1` case. 
Otherwise, we need a count of correct classifications in the current batch